public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -412,13 +411,25 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
       return this.responseParser.parse(query.query, resp);
     });
   }
-
+  
   toMetricFindValue(rsp: DataQueryResponse): MetricFindValue[] {
-    const data = rsp.data ?? [];
+    const valueMap = new Map<string, MetricFindValue>()
     // Create MetricFindValue object for all frames
-    const values = data.map((d) => frameToMetricFindValue(d)).flat();
-    // Filter out duplicate elements
-    return values.filter((elm, idx, self) => idx === self.findIndex((t) => t.text === elm.text));
+    rsp?.data?.forEach((frame: DataFrame) => {
```

Comment:
I think we can also use `reduce` instead of `forEach`. Not sure about the performance impact though. If you wanna check here is the code:

```ts
const valueMap: Map<string, MetricFindValue> = rsp?.data?.reduce((frame: DataFrame, prev) => {
  if (frame && frame.length > 0) {
    let field = frame.fields.find((f) => f.type === FieldType.string);
    if (!field) {
      field = frame.fields.find((f) => f.type !== FieldType.time);
    }
    if (field) {
      field.values.forEach((v) => prev.set(v.toString(), { text: v.toString() }));
    }
  }
  return prev;
}, new Map<string, MetricFindValue>());
```